base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        store=True,
```

Comment:
That was my thought - I generally try to weigh my writes against my reads. Anything >=4x the reads typically gets a stored unless there are specific circumstances to warrant against that.

I think maybe an index is in order though?

One thing that computing it would do - the search could be dynamic against the multiple companies, but as a singleton. Example:

Current:
```
>>> record.id
10
>>> record.company_id.id
1
>>> record.company_ids.ids
[1, 2]
>>> self.search([('company_id', '=', 1), ('id', '=', 10)])
my.model(1)
>>> self.search([('company_id', '=', 2), ('id', '=', 10)])
my.model()
```

With some domain finagling:
```
>>> record.id
10
>>> record.company_id.id
1
>>> record.company_ids.ids
[1, 2]
>>> self.search([('company_id', '=', 1), ('id', '=', 10)])
my.model(1)
>>> self.search([('company_id', '=', 2), ('id', '=', 10)])
my.model(1)
```